primefaces/src/main/java/org/primefaces/application/exceptionhandler/PrimeExceptionHandler.java

```diff
@@ -343,7 +346,17 @@ protected Throwable buildView(FacesContext context, Throwable throwable, Throwab
 
     protected void handleRedirect(FacesContext context, Throwable rootCause, ExceptionInfo info, boolean isResponseReset) throws IOException {
         ExternalContext externalContext = context.getExternalContext();
-        externalContext.getSessionMap().put(ExceptionInfo.ATTRIBUTE_NAME, info);
+        if (externalContext.getSession(false) != null) {
+            ClientWindow clientWindow = externalContext.getClientWindow();
+            if (clientWindow != null && LangUtils.isNotBlank(clientWindow.getId())) {
+                Map<String, ExceptionInfo> windowsMap = (Map<String, ExceptionInfo>)
+                        externalContext.getSessionMap().computeIfAbsent(ExceptionInfo.ATTRIBUTE_NAME + "_map", k -> new ConcurrentHashMap<>());
```

Comment:
we should better use a LimitedSizeHashMap (we already have this in PF) maybe limit up to 5 or 10

in theory we should introduce a LimitedSizeConcurrentHashMap but its unlikely that you parallel access 2 views anyway and this would need to throw a exception in the same millisecond

Reply:
done